### PR TITLE
support TS 4.3 `override` syntax in class

### PR DESCRIPTION
**Pull request by g-plane**
| Q                        | A 
| ------------------------ | ---
| Minor: New Feature?      |Yes
| Tests Added + Pass?      | Yes
| License                  | MIT

Reference: https://devblogs.microsoft.com/typescript/announcing-typescript-4-3-beta/#override-and-the-noimplicitoverride-flag

### CODE REVIEW

**Comment by nicolo-ribaudo** [comment]
Thanks for the PR! However, existentialism is already working on this :sweat_smile: 
https://github.com/babel/babel/issues/13071#issuecomment-812617104

**Comment by g-plane** [comment]
I think we can improve this PR, because it seems that he hasn't submitted any PRs yet.

**Comment by existentialism** [comment]
Totally missed this, I'm fine with letting you finish the PR!

File: packages/babel-parser/src/parser/statement.js [hunk 1]
```diff
@@ -1406,6 +1406,9 @@ export default class StatementParser extends ExpressionParser {
         if (state.hadConstructor && !this.hasPlugin("typescript")) {
           this.raise(key.start, Errors.DuplicateConstructor);
         }
+        if (isConstructor && this.hasPlugin("typescript") && member.override) {
+          this.raise(key.start, Errors.OverrideOnConstructor);
```
Comment (JLHwung) [inline packages/babel-parser/src/parser/statement.js:1410]: TS nightly does not throw on
```ts
class C extends class {} {
  override constructor() { super() };
}
```

https://www.typescriptlang.org/play?ts=4.3.0-dev.20210406#code/MYGwhgzhAEDC0FMAeAXBA7AJjUkYG8BfafAKGmgHsA3BAJzoEtMFphL0IU6BXYFSnQAUAShLQIPAA71R0QgG5ShIA

Comment (g-plane) [reply packages/babel-parser/src/parser/statement.js:1410]: Not right. This is a bug of TypeScript and they fixed it just now: https://github.com/microsoft/TypeScript/commit/f526a38856db6f579beeb883db967963b029f888 .

**Comment by g-plane** [comment]
I will address the rest problems after https://github.com/babel/babel/pull/13096 get merged, then we can share some error messages.

**Comment by g-plane** [comment]
All problems should be addressed.

File: packages/babel-parser/src/plugins/typescript/index.js [hunk 2]
```diff
@@ -91,10 +92,14 @@ const TSErrors = makeErrorTemplates(
       "Index signatures cannot have an accessibility modifier ('%0')",
     IndexSignatureHasDeclare:
       "Index signatures cannot have the 'declare' modifier",
+    IndexSignatureHasOverride:
+      "'override' modifier cannot appear on an index signature.",
     IndexSignatureHasStatic:
       "Index signatures cannot have the 'static' modifier",
     InvalidModifierOnTypeMember:
       "'%0' modifier cannot appear on a type member.",
+    InvalidModifierWithAnotherModifier:
```
Comment (nicolo-ribaudo) [inline packages/babel-parser/src/plugins/typescript/index.js:101]: Nit:
```suggestion
    IncompatibleModifiers:
```

**Review by nicolo-ribaudo — APPROVED** [review]
Thanks!

**Review by existentialism — APPROVED** [review]
Nice work!

**Comment by DanielRosenwasser** [comment]
Just as a heads up, we added a specific order for `override` here: https://github.com/microsoft/TypeScript/pull/43660

**Pull request merged**